File: tc-arg/contest-06/d_digit_sum.py
# TCArg 2020 - Contest #6
# Digit sum [TLE]
# https://codeforces.com/group/j1UosVRZar/contest/287776/problem/D

if __name__ == "__main__":
	a, b = [int(x) for x in input().split()]
	num = a ** b
	while num >= 10:
		total = 0
		while num:
			total += num % 10
			num //= 10
		num = total
	print(total)

File: tc-arg/contest-06/d_digit_sum.cpp
// TCArg 2020 - Contest #6
// Digit sum [WA]
// https://codeforces.com/group/j1UosVRZar/contest/287776/problem/D

#include <iostream>

using namespace std;

int main()
{
	long long a, b, num, total;

	ios::sync_with_stdio(0);
	cin.tie(0);

	cin >> a >> b;
	num = a;
	for (int i = 1; i < b; i++)	{
		num *= a;
		while (num >= 10)	{
			total = 0;
			while (num > 0)	{
				total += num % 10;
				num /= 10;
			}
			num = total;
		}
	}
	cout << total << endl;
	return 0;
}


